pkg/platform/baremetal/template.go

```diff
@@ -70,6 +70,21 @@ module "bare-metal-{{.ClusterName}}" {
 
   ignore_x509_cn_check   = {{.IgnoreX509CNCheck}}
   conntrack_max_per_core = {{.ConntrackMaxPerCore}}
+
+  {{- if .CLCSnippets}}
+  clc_snippets = {
+    {{- range $nodeName, $clcSnippetList := .CLCSnippets }}
+    "{{ $nodeName }}" = [
+    {{- range $clcSnippet := $clcSnippetList }}
+    <<EOF
+{{ $clcSnippet }}
+EOF
```

Comment:
I am not sure if this is correctly linted as per our need. But if it does not add space that it should be linted.

Reply:
We could have unit test for it maybe, importing https://github.com/hashicorp/terraform/blob/master/command/fmt.go, rendering the template and then running a formatter on it :see_no_evil:

Reply:
Surprisingly, it's easy to do:
```go
package main

import (
        "bytes"
        "io/ioutil"
        "os"
        "path/filepath"
        "testing"

        "github.com/hashicorp/terraform/command"
        "github.com/mitchellh/cli"
)

func TestHelloWorld(t *testing.T) {
        //path := "/home/invidian/repos/kinvolk/lokomotive/assets/terraform-modules/tinkerbell-sandbox/outputs.tf"
        path := "/home/invidian/repos/kinvolk/lokomotive/assets/terraform-modules/tinkerbell-sandbox/main.tf"
        content, err := ioutil.ReadFile(path)
        if err != nil {
                t.Fatalf("reading file content: %v", err)
        }

        TerraformFmt(t, content)
}

func TerraformFmt(t *testing.T, content []byte) {
        dir, err := ioutil.TempDir("", "lokoctl-test-")
        if err != nil {
                t.Fatalf("creating temporary directory: %v", err)
        }

        t.Cleanup(func() {
                if err := os.RemoveAll(dir); err != nil {
                        t.Logf("Cleaning temporary directory: %v", err)
                }
        })

        if err := ioutil.WriteFile(filepath.Join(dir, "main.tf"), content, 0666); err != nil {
                t.Fatalf("writing temporary file: %v", err)
        }

        var buf bytes.Buffer
        cmd := command.FmtCommand{
                Meta: command.Meta{
                        RunningInAutomation: true,
                        Ui: &cli.ConcurrentUi{
                                Ui: &cli.BasicUi{
                                        Writer:      &buf,
                                        ErrorWriter: &buf,
                                },
                        },
                },
        }

        if i := cmd.Run([]string{"-check", "-diff", dir}); i != 0 {
                t.Fatalf("Running failed: %v", buf.String())
        }
}
```

Also, make use of https://github.com/hashicorp/terraform/blob/v0.14.4/go.mod#L141-L145 while importing Terraform package.

Reply:
Hm, but it seems importing those packages in Lokomotive will be pain in the ass. I guess we can just fallback to use the binary, expose formatting via `pkg/terraform` and use `e2e` tag for those tests.

Reply:
Created an issue for it https://github.com/kinvolk/lokomotive/issues/1331.

Reply:
This should be tracked as a separate issue. Thanks for creating an issue for it @invidian

Reply:
@ipochi as discussed on Slack, please make sure added code is properly formatted while rendered manually for now and from my side, this comment can be marked as resolved.